doc/terminal-a11y-2023.md

```diff
@@ -0,0 +1,216 @@
+
+This is tracked by [UIA: support movement by page · Issue #13756](https://github.com/microsoft/terminal/issues/13756)
+
+#### UI Automation: Movement by format
+The terminal area supports various text decorations including but not limited to underlined, bold, and italic text. These text attributes were exposed via the `ITextRangeProvider` interface in [PR #10366](https://github.com/microsoft/terminal/pull/10336). UI automation has support for navigating through contiguous spans adhering to a text attribute, which could be useful to expose to UIA clients.
+
+This is tracked by [UIA Formatted Text Navigation · Issue #3567](https://github.com/microsoft/terminal/issues/3567).
+
+#### Movement by prompt
+[PR #12948](https://github.com/microsoft/terminal/pull/12948) added support for scrollbar marks in Windows Terminal. As a part of this, an `experimental.autoMarkPrompts` profile setting registers scrollbar marks with each prompt when enabled. This is a fantastic way to navigate through the scroll history quickly and easily.
+
+Though UI automation doesn't have a text unit for prompts, we could define a "paragraph" as the space between two prompts.
+
+This issue is tracked by [megathread: Scrollbar Marks · Issue #11000](https://github.com/microsoft/terminal/issues/11000).
+
+### Mark Mode support for degenerate range
+[PR #13053](https://github.com/microsoft/terminal/pull/13053) added support for mark mode in Windows Terminal. Mark mode allows users to create and modify selections by exclusively using the keyboard. However, screen reader users have reported it as a strange experience because it always has a cell of text selected; this results in the screen reader reading "x selected, y unselected" as opposed to the expected "x" when moving the cursor around.
+
+Unfortunately, the changes required to fix this are very extensive because selections are stored as two inclusive terminal coordinates, which makes it impossible to represent an empty selection. 
+
+This is tracked by [A11y: windows terminal emits selection/deselection events in mark mode when navigating with arrow keys · Issue #13447](https://github.com/microsoft/terminal/issues/13447).
+
+### Search improvements
+The search dialog can be used to perform a runtime query of the text buffer. However, in its current form, it does not count the total number of matches like other developer tools (i.e. Visual Studio Code). This work item tracks two things:
+1. Support for counting the total number of matches
+2. Displaying the number of results in the search box and announcing it via a UIA notification to the screen reader
+Additional search features may fall into this including but not limited to:
+- Highlight the search results in the scroll bar
+- Successful searches should read the line where text was found (similar to VS Code experience)
```

Comment:
FYI/BTW I benchmarked this and the runtime of `<regex>` practically depends entirely on the number of matches. Given a few thousand matches it takes about 100ms on a slower CPU, which is too much, but not _/too/_ much. `std::boyer_moore_horspool_searcher` would probably be a good alternative by default (>20x faster) if we consider that most of the time people aren't using the regexp search functionality in VS Code either.